### PR TITLE
Open note custom

### PR DESCRIPTION
**Pull request by SiriusXT**
Open notes with custom software.
Use custom software to open notes, and run normally on windows, ubuntu, and manjaro-kde systems.
https://github.com/zadam/trilium/issues/3910

### CODE REVIEW

File: src/public/app/services/open.js [hunk 1]
```diff
@@ -47,6 +47,78 @@ async function openNoteExternally(noteId, mime) {
...
+        const searchTerminal = (index) => {
+          const terminal = terminals[index];
+          if (!terminal) {
+            console.error('Open Note custom: No terminal found!');
+            open(getFileUrl(noteId), { url: true });
+            return;
+          }
+          exec(`which ${terminal}`, (error, stdout, stderr) => {
+            if (stdout.trim()) {
+              foundTerminal = true;
```
Comment (soulsands) [inline src/public/app/services/open.js]: this flag is redundant,  it'd be better to check availableTerminal directly.

File: src/public/app/services/open.js [hunk 2]
```diff
@@ -47,6 +47,78 @@ async function openNoteExternally(noteId, mime) {
...
+      }
+    }
+    else {
+      // allow browser to handle opening common file
+      if (mime === "application/pdf" || mime.startsWith("image") || mime.startsWith("audio") || mime.startsWith("video")) {
+        window.open(getOpenFileUrl(noteId));
+      }
+      else {
+        window.location.href = getFileUrl(noteId);
+      }
```
Comment (soulsands) [inline src/public/app/services/open.js]: this block is repeated. it can be extracted into a new function .

**Review by soulsands — COMMENTED** [review]
you are doing great!

**Comment by zadam** [comment]
Thanks!

**Pull request merged**